File: frontend/src/app/services/config/departments.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ConfigService } from './config.service';
import { Manager } from 'src/app/model/config/manager.model';
import { AppResponse } from 'src/app/dto/response.dto';

@Injectable({
  providedIn: 'root'
})
export class DepartmentsService {


  constructor(private http: HttpClient, private config: ConfigService) { }

  async getManagers(): Promise<AppResponse<Manager[]>> {
    const url = `${this.config.apiUrl}/departments/managers`;
    let data = await fetch(url);
    return await data?.json();
  }
}

File: backend/src/main/java/com/prologiccreations/traderssolution/service/super_classes/config/DepartmentsService.java
package com.prologiccreations.traderssolution.service.super_classes.config;

import com.prologiccreations.traderssolution.dto.Response;
import com.prologiccreations.traderssolution.dto.config.ManagerDto;
import com.prologiccreations.traderssolution.model.config.Departments;
import com.prologiccreations.traderssolution.service.super_classes.CrudService;

import java.util.List;

public interface DepartmentsService extends CrudService<Departments, Long> {

    Response<List<ManagerDto>> findAllManagers();

}
